### PR TITLE
bandcamp_importer: handle IA'd pages

### LINKED ISSUE #322: Feature request: Import Bandcamp pages from Internet Archive
Sometimes the only source for a release is the Internet Archive’s saved Bandcamp page. For example, https://web.archive.org/web/20150215032704/http://myparentsfavoritemusic.bandcamp.com/album/testing-the-waters. I’d like to see an option to be able to easily import these (and good news, the high resolution cover art is often still on the Bandcamp servers!).

### PR DESCRIPTION
**Pull request by vs49688**
Scratches a personal itch of mine.
I should probably note that I am not a JavaScript developer. If there's a better way of doing this, I'd be happy to hear about it.

Fixes #322

### CODE REVIEW

**Comment by vs49688** [comment]
Just ran the linter, error should be fixed now

**Comment by jesus2099** [comment]
I'm curious, what is Wombat?

**Comment by vs49688** [comment]
Wombat is this: https://github.com/webrecorder/wombat

IA uses it for client-side rewriting of URLs to the archive, which conflicts with the Musicbrainz buttons - it rewrites those too, so you end up POST'ing to to `https://web.archive.org/whatever/https://musicbrainz.org`

**Review by kellnerd — COMMENTED** [review]
Thanks, the main import and search button functionality seems to be working. I don't know if there is a better solution, but I've noticed a few things while testing with the example from https://github.com/murdos/musicbrainz-userscripts/issues/322#issue-731904515:
- When I apply only the new `@include` rule, the import button has the old MB icon from 2015, together with the Wombat patch it fails to load the current icon because of CSP.
- The linked MB icons for existing MB artist and release relationships are missing since the underlying MB API calls fail (also because of CSP).

Not sure whether it's really necessary to have all these features on archived pages and whether the required work-arounds are worth the effort. Maybe others have some ideas for improvements, otherwise this PR is good to merge.

**Comment by vs49688** [comment]
> Thanks, the main import and search button functionality seems to be working. I don't know if there is a better solution, but I've noticed a few things while testing with the example from [#322 (comment)](https://github.com/murdos/musicbrainz-userscripts/issues/322#issue-731904515):
> 
>     * When I apply only the new `@include` rule, the import button has the old MB icon from 2015, together with the Wombat patch it fails to load the current icon because of CSP.
> 

Without the Wombat patch, it'll rewrite it to https://web.archive.org/web/20150215032704im_///musicbrainz.org/favicon.ico, so that makes sense.

This is what mine shows:
image

I'm not getting CSP errors for the icon, but I do get a lot of these on Firefox 94.0.
```
Referrer Policy: Less restricted policies, including ‘no-referrer-when-downgrade’, ‘origin-when-cross-origin’ and ‘unsafe-url’, will be ignored soon for the cross-site request: https://musicbrainz.org/favicon.ico
```

It does seem to have issues on the [top-level artist page](https://web.archive.org/web/20150215052226/http://myparentsfavoritemusic.bandcamp.com/) though:




image

>     * The linked MB icons for existing MB artist and release relationships are missing since the underlying MB API calls fail (also because of CSP).
> 

Yup, I'm not sure how to fix these
```
Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://musicbrainz.org/ws/2/url?resource=https%3A%2F%2Fweb.archive.org&inc=artist-rels. (Reason: header ‘x-pywb-requested-with’ is not allowed according to header ‘Access-Control-Allow-Headers’ from CORS preflight response).

Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://musicbrainz.org/ws/2/url?resource=https%3A%2F%2Fweb.archive.org&inc=artist-rels. (Reason: CORS request did not succeed).
```

> 
> Not sure whether it's really necessary to have all these features on archived pages and whether the required work-arounds are worth the effort. Maybe others have some ideas for improvements, otherwise this PR is good to merge.

File: bandcamp_importer.user.js [hunk 1]
```diff
@@ -253,6 +255,39 @@ var BandcampImport = {
     },
 };
 
+window.addEventListener('beforescriptexecute', function (e) {
```
[the remark this answers is not in the document]
Comment (vs49688) [reply bandcamp_importer.user.js]: Done!

**Review by murdos — APPROVED** [review]

**Comment by kellnerd** [comment]
So, I have been using the userscript version from this PR for a while now, I think it's time to finally merge it. The changes make it work on IA pages, have no negative influence on regular Bandcamp pages, and nobody has volunteered to make all the features available on IA pages (which had not even been requested in #322).
Thanks again @vs49688 👍

**Pull request merged**